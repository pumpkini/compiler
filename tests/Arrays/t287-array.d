int main(){
    int i;
    int j;
    int k;
    int[][][] a;

    a = NewArray(3, int[][]);
    for(i = 0; i < 3; i = i+1){
        a[i] = NewArray(i+1, int[]);
    }
    for(i = 0; i < 3; i = i+1){
        for(j = 0; j <= i; j = j+1){
            a[i][j] = NewArray(3, int);
            for(k = 0; k < 3; k = k+1){
                a[i][j][k] = k;
            }
        }
    }

    for(i = 0; i < 3; i = i+1){
        Print("i ", i);
        for(j = 0; j <= i; j = j+1){
            Print("j ", j);
            for(k = 0; k < 3; k = k+1){
                Print(a[i][j][k]);
            }
        }
    }


}