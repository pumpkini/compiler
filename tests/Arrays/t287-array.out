i 0
j 0
0
1
2
i 1
j 0
0
1
2
j 1
0
1
2
i 2
j 0
0
1
2
j 1
0
1
2
j 2
0
1
2
